s3/s3_transport/s3_transport_test.cpp

```diff
@@ -0,0 +1,601 @@
+        }
+
+
+        for (int thread_number = 0; thread_number <  thread_count; ++thread_number) {
+
+            printf("%s:%d (%s) calling join for rw thread %d\n", __FILE__, __LINE__,
+                    __FUNCTION__, thread_number);
+
+            rw_threads[thread_number].join();
+
+            printf("%s:%d (%s) joined wr thread %d\n", __FILE__, __LINE__,
+                    __FUNCTION__, thread_number);
+        }
+        delete[] rw_threads;
+
+    }
+
+    return 0;
+
+}
+
+// to test downloading file to cache
+void open_file_with_read_write(int thread_number,
+                               const int thread_count,
+                               const uint64_t file_size,
+                               const bool debug_flag,
+                               const char *hostname,
+                               const char *bucket_name,
+                               const char * access_key,
+                               const char *secret_access_key,
+                               const char *filename)
+{
+
+    printf("%s:%d (%s) [open file for read/write thread=%u, seq=%u]\n",
+            __FILE__, __LINE__, __FUNCTION__, thread_number, thread_number);
+
+    s3_transport_config s3_config;
+    s3_config.object_size = file_size;
+    s3_config.number_of_transfer_threads = 5;
+    s3_config.bucket_name = bucket_name;
+    s3_config.access_key = access_key;
+    s3_config.secret_access_key = secret_access_key;
+    s3_config.thread_identifier = thread_number;
+    s3_config.debug_flag = debug_flag;
+    s3_config.multipart_flag = use_multipart_flag;
+    s3_config.shared_memory_timeout_in_seconds = 60;
+
+    s3_transport tp1{s3_config};
+    dstream ds1{tp1, s3_prefix + filename};
+
+    if (!ds1.is_open()) {
+        printf("[%d] Open failed.  Exiting...\n", thread_number);
+        return;
+    }
+
+
+    if (thread_number == 0) {
+
+        // test offset write
+        ds1.seekp(0, std::ios_base::end);
+        printf("Adding - adding this to end\n");
+        std::string write_string = "adding this to end\n";
+        ds1.write(write_string.c_str(), write_string.length());
+
+        // test offset read
+        char read_str[21];
+        read_str[20] = 0;
+        printf("------ read 20 bytes at offset 10 -----\n");
+        ds1.seekg(10, std::ios_base::beg);
+        ds1.read(read_str, 20);
+        printf("[%s]\n", read_str);
+        printf("------ read 20 more bytes         -----\n");
+        ds1.read(read_str, 20);
+        printf("[%s]\n", read_str);
+        printf("------ read 20 more bytes skip 10 -----\n");
+        ds1.seekg(10, std::ios_base::cur);
+        ds1.read(read_str, 20);
+        printf("[%s]\n", read_str);
+        printf("------ read last 20 bytes         -----\n");
+        ds1.seekg(-20, std::ios_base::end);
+        ds1.read(read_str, 20);
+        printf("[%s]\n", read_str);
+        printf("---------------------------------------\n");
+    }
+
+    // will be automatic
+    //ds1.close();
```

Comment:
dead

Reply:
This program is no longer used for unit testing and has been removed.